crypto/stark-verifier-ethereum/contracts/fri.sol

```diff
@@ -29,20 +83,203 @@ contract Fri {
                 }
             }
         }
-        sort(queries);
+        queries.sort();
         return queries;
     }
 
-    // This function sorts the array
-    // Note - We use insertion sort, the array is expected to be small so this shouldn't
-    // cause problems.
-    function sort(uint64[] memory data) internal pure {
-        for (uint256 i = 0; i < data.length; i++) {
-            uint256 j = i;
-            while (j > 0 && data[j] < data[j - 1]) {
-                (data[j], data[j - 1]) = (data[j - 1], data[j]);
-                j--;
+    // Unwraping endpoint because the main function has too deep of a stack otherwise
+    function fri_check(
+        ProofTypes.StarkProof memory proof,
+        uint8[] memory fri_layout,
+        uint256[] memory eval_points,
+        uint8 log_eval_domain_size,
+        uint64[] memory queries,
+        uint256[] memory polynomial_at_queries,
+        uint256 oods_point,
+        uint256 evaluated_oods_point
+    ) internal {
+        fold_and_check_fri_layers(
+            FriContext(
+                proof.fri_values,
+                proof.fri_commitments,
+                proof.fri_decommitments,
+                fri_layout,
+                eval_points,
+                log_eval_domain_size,
+                queries,
+                polynomial_at_queries,
+                proof.last_layer_coeffiencts
+            )
+        );
+
+        // The final check is that the constraints evaluated at the out of domain sample are
+        // equal to the values commited constraint values
```

Comment:
I don't think the oods check should be part of the FRI. All FRI is supposed to do is check that the commitment is to a low degree polynomial